### ISSUE TITLE
patches applied before fuseki server is started

### ISSUE THREAD

**Issue by bsara** [afs/rdf-delta#103]
I'm having a problem because of the fact that the fuseki server is not started before unapplied patches are fetched and applied at server startup time.

Here is the problem:
I'm running the rdf-delta fuseki server in a docker container within a k8s cluster. k8s determines the health of the container by hitting `/$/ping` of the fuseki server. However, when a patch (or many patches) are so large that they take a long time to sync during startup, the fuseki server takes too long to start, which causes k8s to think that there is something wrong with the server and it is automatically restarted. I've ended up in an infinite bootloop with one of my servers because of this. It seems that it would make more sense to make the fuseki server available as soon as possible so that at least reading can take place on the server immediately.

Basically, I'm requesting that the fuseki server be started immediately on startup rather than waiting to sync any unsynced changes and start syncing the changes once the server has triggered to start.

**Comment by afs** [comment]
Seems reasonable if out-of-date data is acceptable - it does not look like it is a simple matter to separate dataset building from the Fuseki admin operations without needing an exclusion lock on the dataset which in turn risks other things being held up.

For background information, how big are the patches a dn howlong the backlog that arises in your deployment?
Is this TDB1 or TDB2?

**Comment by bsara** [comment]
using TDB2 and the patches are around 5 MB each in this case.

I figured, patches automatically get synced when a new request is submitted and the server is already running, so, it shouldn't be too detrimental (if at all) to do the same after the server starts up.

**Comment by afs** [comment]
5MB is maybe ~500K changes. That shouldn't take very long to apply if there is only one patch outstanding (is that the case here?)

How quickly is the k8s taking before declaring the server inoperative? maybe it is being too impatient!

It might be better to fork the initial sync so the server continues to start as normal. In fact, have that as a general facility feels like a good idea.

**Comment by bsara** [comment]
> It might be better to fork the initial sync so the server continues to start as normal. In fact, have that as a general facility feels like a good idea.

Can you elaborate on what you mean by this?

**Comment by afs** [comment]
The idea is to run a background regular sync attempt that runs after 0 seconds then every 5 minutes.

Currently, this impacts a test suite - tests run in isolation but when run immediate one after another, the second test fails (whichever test is second; changing the order in suite changes which fails). 

This isn't too bad - the tests are not normal usage patterns. They start and stop servers very quickly, and the fact some network clear-up (the port+socket by the OS) doesn't happen fast enough is not a new situation.
